scipy/stats/tests/test_stats.py

```diff
@@ -764,6 +770,24 @@ def test_nd_special_cases(self, xp):
         xp_assert_close(ci.low, -ones)
         xp_assert_close(ci.high, ones)
 
+    def test_different_dimensionality(self, xp):
```

Comment:
I tried this test with 1.14.1 and it failed, is this expected?

Reply:
Yup. I noticed while I was addressing this that the code was inconsistent. That's why I swapped the order of things. This opens up behavior that failed before (that shouldn't have), but it doesn't prevent anything that did work before. Try the test with basically any other two-arg reducing function.